edn/src/types.rs

```diff
@@ -41,15 +41,74 @@ pub enum Value {
 
 use self::Value::*;
 
-impl Value {
-    pub fn is_keyword(&self) -> bool {
-        match *self {
-            Keyword(_) => true,
-            _          => false,
+macro_rules! def_is {
+    ($name: ident, $kind: pat) => {
+        pub fn $name(&self) -> bool {
+            match *self { $kind => true, _ => false }
         }
     }
 }
 
+impl Value {
+    def_is!(is_nil, Nil);
+    def_is!(is_boolean, Boolean(_));
+    def_is!(is_integer, Integer(_));
+    def_is!(is_big_integer, BigInteger(_));
+    def_is!(is_float, Float(_));
+    def_is!(is_text, Text(_));
+    def_is!(is_symbol, PlainSymbol(_));
+    def_is!(is_namespaced_symbol, NamespacedSymbol(_));
+    def_is!(is_keyword, Keyword(_));
+    def_is!(is_namespaced_keyword, NamespacedKeyword(_));
+    def_is!(is_vector, Vector(_));
+    def_is!(is_list, List(_));
+    def_is!(is_set, Set(_));
+    def_is!(is_map, Map(_));
+
+    pub fn as_nil(&self) -> Option<Value> {
```

Comment:
Probably don't need `as_nil` when we have `is_nil`, but you could do `Option<()>`, which is really just success or failure.